src/bootstrap/compile.rs

```diff
@@ -655,7 +677,6 @@ impl Step for CodegenBackend {
         }
 
         let out_dir = builder.cargo_out(compiler, Mode::Codegen, target);
-        builder.clear_if_dirty(&out_dir, &librustc_stamp(builder, compiler, target));
```

Comment:
Hm, is this change required? I'd expect that if we change rustc crates we need this in order for the codegen backends to be properly rebuilt.